feat: migrate project UI to dashboard and related pages

alok0814 opened the pull request:
## 概要
- 提案したUIを`/dashboard`ページおよび関連ページに移植しました。
- Tailwind/PostCSS/Next.jsの設定も統合し、App Router構成に対応。

## 主な変更点
- `src/app/dashboard/page.tsx`を新UIで置き換え
- `components`・`lib`・`analytics`・`customers`・`settings`などの移植
- tailwind.config.js, postcss.config.mjs, next.config.mjsのマージ
- importパスや依存の修正

## 注意点・未対応事項
- 一部のAPIやDB連携はモックデータのままです
- 顔認証・登録機能は別途実装・調整が必要
- スタイルやレイアウトの微調整は今後も発生する可能性あり

## 動作確認方法
1. `cd web && npm install`
2. `npm run dev`
3. `http://localhost:3000/dashboard` で表示されなかった...

## その他
- 追加のレビュー観点や質問があればコメントください

alok0814 committed:
feat: migrate project UI to dashboard and related pages

alok0814 committed:
fix: mainブランチとのコンフリクトを解消

Harineko0 commented on package.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "devDependencies": {
+    "@tailwindcss/postcss": "^4.1.10"
+  }
+}
```
must: 依存ライブラリのインストールは `web/` と `backend/` のディレクトリ内で `pnpm install` を実行する形でお願いします 🙏 
あと、`node_modules/` が Git に含まれちゃってるので削除お願いします

alok0814 replied:
対応済み: READMEにweb/とbackend/でpnpm installを実行する形で明記
対応済み: node_modules/をGitから削除し、.gitignoreに追加

Harineko0 commented on web/src/app/customers/page.tsx:
```diff
@@ -0,0 +1,16 @@
+  },
+}
+
+export default function Page() {
```
imo: 店舗側が見れるページは全て `dashboard/` から始まるようにすると middleware の設定的にもやりやすいかなーと思うのですが、いかがでしょうか？

alok0814 replied:
READMEにweb/とbackend/でpnpm installを実行する手順を追記し、node_modules/もGit管理から外しました。
ご確認よろしくお願いします！

alok0814 replied:
店舗側のページはdashboard/配下にまとめるようにリファクタしました。
middlewareの設定もシンプルになったと思います。

Harineko0 commented on web/src/app/sitemap.ts:
```diff
@@ -0,0 +1,44 @@
+import { mockCustomers } from '@/lib/data'
+
+export default function sitemap(): MetadataRoute.Sitemap {
+  const baseUrl = 'https://bar-cms.example.com'
```
must: baseUrl は環境変数 or headers() から取得するといいかと思いますー

alok0814 replied:
READMEにweb/とbackend/でpnpm installを実行する手順を追記し、node_modules/もGit管理から外しました。
ご確認よろしくお願いします

alok0814 replied:
baseUrlの取得方法を環境変数（NEXT_PUBLIC_API_URLやAPI_URL）から取得する形に修正しました

Harineko0 commented on web/src/lib/types.ts:
```diff
@@ -0,0 +1,36 @@
+export interface Customer {
```
must: ドメインの型定義は openapi/types.ts を使うようといいかと思います！

alok0814 replied:
ドメインの型定義をopenapi/types.tsを参照するように修正しました！
型の重複がないように整理しています、、、

Harineko0 commented on web/src/lib/data.ts:
```diff
@@ -0,0 +1,92 @@
+import { Customer } from './types';
+
+export const mockCustomers: Customer[] = [
```
want: customers を取得するエンドポイントは `api/v1/stores/{storeId}/customers` に追加したので、 #86 と同様に `web/src/api.ts` を用いて呼び出しお願いします 🙏

alok0814 replied:
customersの取得もweb/src/api.ts経由でAPI呼び出しするように修正しました。
#86の実装に合わせています。

Harineko0 reviewed (commented):
ざっと目を通しましたー
UI の見た目は Preview 環境がデプロイできたら見ます
#87 は #86 のブランチから生やしたと思うので、 #86 を先にマージしたいですね

alok0814 committed:
chore: remove node_modules and update .gitignore

alok0814 committed:
refactor: move customers page to dashboard structure

alok0814 committed:
refactor: use openapi types for API communication and fix baseUrl configuration

alok0814 committed:
docs: update installation instructions to use pnpm in web/ and backend/ directories

alok0814 committed:
fix: remove vector.ts to resolve CI build errors

alok0814 committed:
Merge branch 'main' into feature/migrate-project-ui

alok0814 committed:
Fix ESLint errors and Prettier warnings

alok0814 committed:
Fix ESLint errors in web app and update configurations

alok0814 committed:
Merge conflicts resolved

alok0814 committed:
Fix API URL configuration and other improvements

alok0814 committed:
Merge main into feature/migrate-project-ui - resolve conflicts with #86 changes

alok0814 committed:
fix: resolve TypeScript build error in face-embedding-repo.ts

alok0814 committed:
chore: fix code style issues with Prettier

alok0814 committed:
chore: fix lint and format errors

alok0814 committed:
feat: resolve conflicts by accepting all changes from main

alok0814 committed:
fix: correct type definition in customer detail page

alok0814 committed:
fix: correct type definition in customer detail page

alok0814 committed:
Merge main branch with conflicts resolved (main priority)

alok0814 committed:
Resolve stash conflicts with main priority

alok0814 committed:
Fix build issues and resolve conflicts with main branch

- Fix TypeScript syntax error in face-embedding-repo.ts
- Update web dependencies (pnpm-lock.yaml)
- All projects now build successfully:
  - Web (Next.js): Static build completed
  - Backend (TypeScript): Compilation successful
  - AI (Python): Dependencies installed and server running
